### PR TITLE
Fix timezone dependence 

### PR DESCRIPTION
**Pull request by mbrandonw**
While out here in Cape Town I realized that some tests fail locally here even though they pass on circleci. The culprit is time zones of course! The fix was pretty easy.

We currently have a redundancy in what we consider to be the current global timezone. We have an environment variable for the timezone, but we also have an environment var for the calendar which also has a timezone! _Sacrebleu!_

The fix is just to get rid of the timezone global and only ever use the timezone in the calendar if we need to access it. 

This simplifies things quite a bit. I made a few other changes that I'll comment on too...

### CODE REVIEW

File: Kickstarter-iOS/Tests/Views/FundingGraphViewTests.swift [hunk 1]
```diff
@@ -92,6 +92,34 @@ internal final class FundingGraphViewTests: TestCase {
       FBSnapshotVerifyView(graphView, identifier: "state_\(key)")
     }
   }
+
+  func testOneDayProject() {
```
Comment (mbrandonw) [inline Kickstarter-iOS/Tests/Views/FundingGraphViewTests.swift]: added a new screenshot test for weird, edge-case one day projects cause i started getting worried i was gonna mess something up with the funding graph.

/cc @luoser

File: Kickstarter-iOS/Views/FundingGraphView.swift [hunk 2]
```diff
@@ -255,30 +255,6 @@ private func slope(ofLine line: Line) -> CGFloat {
 private func dateToDayNumber(launchDate: TimeInterval,
```
Comment (mbrandonw) [inline Kickstarter-iOS/Views/FundingGraphView.swift]: @luoser turns out your `dateToDayNumber` and `totalNumberOfDays` do the same thing, so i removed one, and the definition of the function could be simplified.

File: Library/Format.swift [hunk 3]
```diff
@@ -98,7 +98,7 @@ public enum Format {
         dateStyle: dateStyle,
         locale: env.locale,
         timeStyle: timeStyle,
-        timeZone: env.timeZone
+        timeZone: env.calendar.timeZone
```
Comment (mbrandonw) [inline Library/Format.swift]: just use the calendar's timezone if ya need it!

File: Library/Tests/TestHelpers/TestCase.swift [hunk 4]
```diff
@@ -23,11 +23,16 @@ internal class TestCase: FBSnapshotTestCase {
 
   override func setUp() {
     super.setUp()
+
+    var calendar = Calendar(identifier: .gregorian)
+    calendar.timeZone = TimeZone(identifier: "GMT")!
```
Comment (mbrandonw) [inline Library/Tests/TestHelpers/TestCase.swift]: previously tests were using `Calendar.current` which could mean people running tests with their computer set to a different calendar could get failing tests.

Of course, Im not sure who uses different calendars on their computers... seems wild...

File: Library/Format.swift (hunk 3 above)
Comment (theginbin) [inline Library/Format.swift]: sweet!

File: Kickstarter-iOS/Views/FundingGraphView.swift (hunk 2 above)
Comment (theginbin) [inline Kickstarter-iOS/Views/FundingGraphView.swift]: it might be nice to rename this swift 3 style, something like `days(from launchDate: Date, to currentDate: Date)`, but doubt you want to do that now...

Comment (luoser) [inline Kickstarter-iOS/Views/FundingGraphView.swift]: grazi

**Comment by luoser** [comment]
lol @theginbin
